kfs-core/src/main/java/edu/arizona/kfs/fp/document/validation/impl/DisbursementVoucherNonEmployeeTravelValidation.java

```diff
@@ -0,0 +1,214 @@
+			errors.putError(KFSPropertyConstants.DISB_VCHR_TRAVEL_TO_STATE_CODE, KFSKeyConstants.ERROR_DV_TRAVEL_TO_STATE);
+			isValid = false;
+		}
+
+		if (!isValid) {
+			errors.removeFromErrorPath(KFSPropertyConstants.DV_NON_EMPLOYEE_TRAVEL);
+			errors.removeFromErrorPath(KFSPropertyConstants.DOCUMENT);
+			return false;
+		}
+
+		/* must fill in all required per diem fields if any field is filled in */
+		boolean perDiemSectionComplete = validatePerDiemSection(document, errors);
+
+		/*
+		 * must fill in all required personal vehicle fields if any field is
+		 * filled in
+		 */
+		boolean personalVehicleSectionComplete = validatePersonalVehicleSection(document, errors);
+
+		/*
+		 * make sure per diem fields have not changed since the per diem amount
+		 * calculation
+		 */
+		if (!perDiemSectionComplete) {
+			isValid = false;
+		}
+
+		// validate the tax amount
+		isValid &= validateTravelAmount(document);
+
+		/*
+		 * make sure mileage fields have not changed since the mileage amount
+		 * calculation
+		 */
+		if (personalVehicleSectionComplete) {
+			KualiDecimal currentCalcAmt = document.getDvNonEmployeeTravel().getDisbVchrMileageCalculatedAmt();
+			KualiDecimal currentActualAmt = document.getDvNonEmployeeTravel().getDisbVchrPersonalCarAmount();
+			if (ObjectUtils.isNotNull(currentCalcAmt) && ObjectUtils.isNotNull(currentActualAmt)) {
+				KualiDecimal calculatedMileageAmount = getDisbursementVoucherTravelService().calculateMileageAmount(document.getDvNonEmployeeTravel().getDvPersonalCarMileageAmount(), document.getDvNonEmployeeTravel().getDvPerdiemStartDttmStamp());
+				if (calculatedMileageAmount.compareTo(document.getDvNonEmployeeTravel().getDisbVchrMileageCalculatedAmt()) != 0) {
+					errors.putErrorWithoutFullErrorPath(KFSConstants.GENERAL_NONEMPLOYEE_TAB_ERRORS, KFSKeyConstants.ERROR_DV_MILEAGE_CALC_CHANGE);
+					isValid = false;
+				}
+
+				// determine if the rule is flagged off in the parm setting
+				boolean performTravelMileageLimitInd = parameterService.getParameterValueAsBoolean(DisbursementVoucherDocument.class, DisbursementVoucherConstants.NONEMPLOYEE_TRAVEL_ACTUAL_MILEAGE_LIMIT_PARM_NM);
+				if (performTravelMileageLimitInd) {
+					// if actual amount is greater than calculated amount
+					if (currentCalcAmt.subtract(currentActualAmt).isNegative()) {
+						errors.putError(KFSPropertyConstants.DV_PERSONAL_CAR_AMOUNT, KFSKeyConstants.ERROR_DV_ACTUAL_MILEAGE_TOO_HIGH);
+						isValid = false;
+					}
+				}
+			}
+		}
+		
+		if (document.getDvNonEmployeeTravel().getDvPerdiemStartDttmStamp() != null || document.getDvNonEmployeeTravel().getDvPerdiemEndDttmStamp() != null) {
+			if (document.getDvNonEmployeeTravel().getDvPerdiemStartDttmStamp().compareTo(document.getDvNonEmployeeTravel().getDvPerdiemEndDttmStamp()) >= 0) {
+				LOG.info("inside last if statement");
```

Comment:
This looks to be more along the lines of developer debug logging. Can this be removed or is this statement beneficial for a production environment?